core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -576,19 +576,13 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       throw new ConfigException(s"Disabling the '${GroupType.CLASSIC}' protocol is not supported.")
     }
     if (protocols.contains(GroupType.CONSUMER)) {
-      if (processRoles.isEmpty) {
-        throw new ConfigException(s"The new '${GroupType.CONSUMER}' rebalance protocol is only supported in KRaft cluster.")
-      }
-      if (!isNewGroupCoordinatorEnabled) {
-        throw new ConfigException(s"The new '${GroupType.CONSUMER}' rebalance protocol is only supported by the new group coordinator.")
+      if (processRoles.isEmpty || !isNewGroupCoordinatorEnabled) {
+        warn(s"The new '${GroupType.CONSUMER}' rebalance protocol is only supported in KRaft cluster with the new group coordinator.")
```

Comment:
Maybe `KafkaAPIs#isConsumerGroupProtocolEnabled` needs similar log when the `isConsumerRebalanceProtocolSupported` return false and `CONSUMER` protocol is enabled.

https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L3816

Reply:
Hum... This is too much in my opinion because it would log on every calls from the consumers. Having a warning at the beginning seems enough for me.

Reply:
> it would log on every calls from the consumers

you are right :100: